### PR TITLE
Added documentation about callable middleware.

### PR DESCRIPTION
**Pull request by roaldnefs**
Added documentation to the middleware section explaining how middleware factories are made callable.

### CODE REVIEW

File: docs/topics/http/middleware.txt [hunk 1]
```diff
@@ -90,6 +90,17 @@ caveats:
 * Unlike the ``__call__()`` method which is called once per request,
   ``__init__()`` is called only *once*, when the Web server starts.
 
+``__call__(request)``
+---------------------
+
+Middleware factories are made callable by implementing the ``__call__()``
+method and must accept a ``request`` argument and return a response. The
+response can be retrieved by calling its ``get_response``. If the
+``__call__()`` method returns a response without every calling its
```
Comment (jacobtylerwalls) [inline docs/topics/http/middleware.txt:99]: "without ever"

File: docs/topics/http/middleware.txt [hunk 2]
```diff
@@ -90,6 +90,17 @@ caveats:
...
+``get_response``, none of the following middleware layers will see the request
+or the response. In this case the response will only return through the same
+middleware layers the request passed in though.
```
Comment (jacobtylerwalls) [inline docs/topics/http/middleware.txt:102]: "passed in through"

**Review by jacobtylerwalls — COMMENTED** [review]
Spotted some minor typos. Cheers, Jacob

File: docs/topics/http/middleware.txt (hunk 1 above)
Comment (roaldnefs) [reply docs/topics/http/middleware.txt]: Thanks @jacobtylerwalls for your suggestion! I've applied it in 5f89049.

File: docs/topics/http/middleware.txt (hunk 2 above)
Comment (roaldnefs) [reply docs/topics/http/middleware.txt]: Applied your suggestion in 5f89049.

**Review by carltongibson — COMMENTED** [review]
I'm going to close this I think. For me, the existing `SimpleMiddleware` example above does everything of this added paragraph but more clearly (with code). 

* I think it's reasonable for us to expect folks to know what a callable is, or to search for that. 
* The `__init__()` notes add value, since it's likely not otherwise obvious that the middleware are instantiated once at startup. 

Thanks @roaldnefs

**Pull request closed without merging**